examples/rust/custom_space_view/src/color_coordinates_view_part_system.rs

```diff
@@ -42,12 +43,11 @@ impl NamedViewSystem for InstanceColorSystem {
 }
 
 impl ViewPartSystem for InstanceColorSystem {
-    /// The archetype this scene part is querying from the store.
-    ///
-    /// TODO(wumpf): In future versions there will be a hard restriction that limits the queries
-    ///              within the `populate` method to this archetype.
-    fn archetype(&self) -> ArchetypeDefinition {
-        ColorArchetype::all_components().try_into().unwrap()
+    fn required_components(&self) -> IntSet<ComponentName> {
+        ColorArchetype::required_components()
```

Comment:
Should `required_components` just return an IntSet natively?

Reply:
I want to keep these core methods allocation-free but... maybe we can generate a static intset in addition to the static array... hmm, I'll have a look.

Reply:
Eeeeh no actually there's still all the mesh stuff to ship so I'll just leave a comment for now, but yeah something to keep in mind